pkg/sql/logictest/testdata/logic_test/datetime

```diff
@@ -471,7 +471,12 @@ SELECT '2015-08-25 05:45:45.53453'::timestamptz
 query T
 SELECT '2015-08-25 05:45:45-01:00'::timestamp
 ----
-2015-08-25 06:45:45 +0000 +0000
+2015-08-25 05:45:45 +0000 +0000
+
+query T
+SELECT '2015-08-25 05:45:45-01:00'::timestamp::timestamptz
+----
+2015-08-25 07:45:45 +0100 +0100
```

Comment:
huh, this one does not match postgres:

```
david=# set time zone +1;
SET
david=# SELECT '2015-08-25 05:45:45-01:00'::timestamp::timestamptz;
      timestamptz
------------------------
 2015-08-25 05:45:45+01
(1 row)
```

Reply:
what do we think is the correct behaviour? granted I am not quite the timestamp-knower that dan seems to have me pegged as now, but I find cockroach's behaviour in this logic test surprising, given the one above it. it seems like the only way this behaviour should happen is if the intermediate `timestamp` knows which time zone it was rendered from (despite not presenting that info) is that how `timestamp`s work?

Reply:
so one big difference in behavior from pg is this:
```
In a literal that has been determined to be timestamp without time zone, PostgreSQL will silently ignore any time zone indication. That is, the resulting value is derived from the date/time fields in the input value, and is not adjusted for time zone.
```

I'm very wary of that, but in an example like this, i'm not sure we'll be able to match their result without doing so.

The other is that when pg casts a timestamp to a timestamptz, it appears to just assume it was already in the session offset, i.e. the rendered hour doesn't change, but now has an offset... if you'd assumed the original `timestamp` had a 0 offset, that would mean the utc/unix time of the underlying time.Time has changed.

Huh. I don't like either of those.

Reply:
indeed, pg docs say:
> Conversions between timestamp without time zone and timestamp with time zone normally assume that the timestamp without time zone value should be taken or given as timezone local time. A different time zone can be specified for the conversion using AT TIME ZONE.